src/error.rs

```diff
@@ -262,6 +269,85 @@ fn from_sslerror_ref(err: &ssl::Error) -> Error {
     }
 }
 
+#[cfg(feature = "snappy")]
+fn from_snap_error_ref(err: &::snap::Error) -> Error {
+    match err {
+        &::snap::Error::TooBig { given, max } => {
```

Comment:
Well even `Box<Error>` makes it completely opaque, which I wouldn't consider ok as a library either. I'll go ahead and make an issue to track switching to `error_chain`.